gatherling/models/Database.php

```diff
@@ -14,29 +14,38 @@ public static function getConnection()
 
         if (!isset($instance)) {
             global $CONFIG;
-            $instance = new mysqli(
+            $instance_attempt = new mysqli(
                 $CONFIG['db_hostname'],
                 $CONFIG['db_username'],
                 $CONFIG['db_password']
             );
             if (mysqli_connect_errno()) {
-                echo mysqli_connect_error();
-                exit("\nfailed to connect to mysql server");
+                if (PHP_SAPI == 'cli') {
+                    // When running the db-upgrade script, we want to return null
+                    // so that it can retry connections as needed.
+                    return null;
```

Comment:
If there are other CLI tools (outside `db-upgrade.php`), they'll need to handle the nullable return type from `getConnection()`. This was necessary because db-upgrade.php needs to be able to retry connections.

Also fun fact that `exit("some string");` gives you an exit code of 0, which is why I replaced it with `exit(1);` for error scenarios.